source/agora/consensus/EnrollmentManager.d

```diff
@@ -797,6 +797,67 @@ public class EnrollmentManager
         return true;
     }
 
+    /***************************************************************************
+
+        Get the pre-image hash for this validator
+
+        Params:
+            enroll_key = UTXO enrollemnt key
+            height = the block height of pre-image to be returned
+
+        Returns:
+            The preimage hash if found otherwise Hash.init
+
+    ***************************************************************************/
+
+    public Hash getPreimage (in Height height) @safe nothrow
+    {
+        return this.cycle[height];
+    }
+
+    /***************************************************************************
+
+        Get the pre-image hash for a validator by either the public key or the
+        enrollment key and height
+
+        Params:
+            public_key = validator public key
+            enroll_key = UTXO enrollemnt key
+            height = the block height of pre-image to be returned
+
+        Returns:
+            The preimage hash if found otherwise Hash.init
+
+    ***************************************************************************/
+
+    public Hash getValidatorPreimageAt (in PublicKey public_key, in Height height) @safe nothrow
+    {
+        return this.getValidatorPreimageAt(getEnrollmentForKey(public_key), height);
+    }
+
+    /// Ditto
+    public Hash getValidatorPreimageAt (in Hash enroll_key, in Height height) @safe nothrow
+    {
+        return this.validator_set.getPreimageAt(enroll_key, height).hash;
+    }
```

Comment:
We already have way too many one liners which forward, let's not add some.
Most of those one liner give a slightly different piece of data, I'd really wish we consolidate them.

Reply:
I'll drop the second one as it is only being used here.